### PR TITLE
possibility to sent survey for previous months

### PR DESCRIPTION
**Pull request by martarzepa**

### CODE REVIEW

File: backend/resources/surveys.py [hunk 1]
```diff
@@ -458,8 +460,8 @@ def post(self, survey_id):
             abort(403)
 
         # Check if there is no answers for this team in this period
-        if team.answered():
-            abort(400, 'Only one answer per team is allowed.')
+#         if team.answered():
+#             abort(400, 'Only one answer per team is allowed.')
```
Comment (marcin992) [inline backend/resources/surveys.py]: Please uncomment it

File: frontend/src/components/SurveyPage/SurveyPage.jsx [hunk 2]
```diff
@@ -1,80 +1,133 @@
...
 import Survey from "./Survey";
 import TemplatePage from "../common/TemplatePage/";
 import Loader from "./../common/Loader/";
+import {DateTime} from "luxon";
+import { setUserTribesDetails } from "../../store/actions/user";
+import { setTribeMatrix, setTribePeriods } from "../../store/actions/results";
 
-const SurveyPage = ({ user, teams, currentTeamId, surveyIsActive, ...props }) => {
-  const [isLoading, setIsLoading] = useState(false);
+const SurveyPage = ({ user, teams, currentTeamId, surveyIsActive, tribes, periods,  ...props }) => {
```
Comment (marcin992) [inline frontend/src/components/SurveyPage/SurveyPage.jsx]: Set indentation to 2

Comment (marcin992) [inline frontend/src/components/SurveyPage/SurveyPage.jsx]: `tribes` aren't used, you can remove it from props

File: frontend/src/components/SurveyPage/SurveyPage.jsx [hunk 3]
```diff
@@ -1,80 +1,133 @@
...
-    let promises = [
-      props.setCurrentSurveyTeamId(teams[0].id),
-      props.setTeamAnswers(teams[0].id),
-      props.setCurrentSurvey(teams[0].tribe_id)
-    ];
-    Promise.all(promises).then(() => setIsLoading(false));
-  }
+    useEffect(() => {
+        if (tribes[0]) handleTribeSelect(null, { value: tribes[0].id });
+    }, [tribes]);
```
Comment (marcin992) [inline frontend/src/components/SurveyPage/SurveyPage.jsx]: You can remove it

File: frontend/src/components/SurveyPage/SurveyPage.jsx [hunk 4]
```diff
@@ -1,80 +1,133 @@
...
+        ];
+        Promise.all(promises).then(() => setIsLoading(false));
+    };
+
+    const handleTribeSelect = (e, { value }) => {
+        if (value === currentTribeId) return;
+        setCurrentTribeId(value);
+        fetchMatrix(value);
+        props.setTribePeriods(value);
+    };
```
Comment (marcin992) [inline frontend/src/components/SurveyPage/SurveyPage.jsx]: You can remove it

File: frontend/src/components/SurveyPage/SurveyPage.jsx [hunk 5]
```diff
@@ -1,80 +1,133 @@
...
 const mapStateToProps = state => ({
-  user: state.user.userData,
-  teams: state.user.teams || [],
-  currentTeamId: state.currentSurvey.team_id,
-  surveyIsActive: !state.results.team.length
+    user: state.user.userData,
+    teams: state.user.teams || [],
+    currentTeamId: state.currentSurvey.team_id,
+    surveyIsActive: !state.results.team.length,
+    tribes: state.user.tribes || [],
```
Comment (marcin992) [inline frontend/src/components/SurveyPage/SurveyPage.jsx]: You can remove it

File: backend/resources/surveys.py [hunk 6]
```diff
@@ -487,7 +489,14 @@ def post(self, survey_id):
 
         # Answer objects list
         answer_obj = []
-        today = date.today()
+
+        if 'period_id' in args:
+            period = Period.query.filter_by(id=args['period_id'], tribe_id=team.tribe_id).one()
+            print (period)
```
Comment (marcin992) [inline backend/resources/surveys.py]: Remove print

**Review by marcin992 — CHANGES_REQUESTED** [review]

File: frontend/src/components/SurveyPage/SurveyPage.jsx (hunk 2 above)
Comment (martarzepa) [reply frontend/src/components/SurveyPage/SurveyPage.jsx]: done

Comment (martarzepa) [reply frontend/src/components/SurveyPage/SurveyPage.jsx]: done

File: frontend/src/components/SurveyPage/SurveyPage.jsx (hunk 3 above)
Comment (martarzepa) [reply frontend/src/components/SurveyPage/SurveyPage.jsx]: done

File: frontend/src/components/SurveyPage/SurveyPage.jsx (hunk 4 above)
Comment (martarzepa) [reply frontend/src/components/SurveyPage/SurveyPage.jsx]: done

File: frontend/src/components/SurveyPage/SurveyPage.jsx (hunk 5 above)
Comment (martarzepa) [reply frontend/src/components/SurveyPage/SurveyPage.jsx]: done

**Review by marcin992 — APPROVED** [review]

**Pull request merged**